src/Reflection/Annotations/AnnotationsMethodsClassReflectionExtension.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Annotations;
+
+use PHPStan\Broker\Broker;
+use PHPStan\Reflection\BrokerAwareClassReflectionExtension;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use PHPStan\Type\FileTypeMapper;
+use PHPStan\Type\VoidType;
+
+class AnnotationsMethodsClassReflectionExtension implements MethodsClassReflectionExtension, BrokerAwareClassReflectionExtension
+{
+
+	/** @var FileTypeMapper */
+	private $fileTypeMapper;
+
+	/** @var Broker */
+	private $broker;
+
+	/** @var MethodReflection[][] */
+	private $methods = [];
+
+	public function __construct(FileTypeMapper $fileTypeMapper)
+	{
+		$this->fileTypeMapper = $fileTypeMapper;
+	}
+
+	public function setBroker(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function hasMethod(ClassReflection $classReflection, string $methodName): bool
+	{
+		if (!isset($this->methods[$classReflection->getName()])) {
+			$this->methods[$classReflection->getName()] = $this->createMethods($classReflection);
+		}
+
+		return isset($this->methods[$classReflection->getName()][$methodName]);
+	}
+
+	public function getMethod(ClassReflection $classReflection, string $methodName): MethodReflection
+	{
+		return $this->methods[$classReflection->getName()][$methodName];
+	}
+
+	/**
+	 * @param ClassReflection $classReflection
+	 * @return MethodReflection[]
+	 */
+	private function createMethods(ClassReflection $classReflection): array
+	{
+		$methods = [];
+		foreach ($classReflection->getParentClassesNames() as $parentClassName) {
```

Comment:
You can iterate over `getParentClasses()` which gives you reflection objects, so you don't have to write that `if` with `hasClass`.